equinix/data_source_network_device.go

```diff
@@ -0,0 +1,906 @@
+package equinix
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/equinix/ne-go"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+var deviceStates = []string{ // Not sure if other states should be included
+	// ne.DeviceStateClusterSetUpInProgress,
+	// ne.DeviceStateDeprovisioned,
+	// ne.DeviceStateDeprovisioning,
+	// ne.DeviceStateFailed,
+	// ne.DeviceStateInitializing,
+	ne.DeviceStateProvisioned,
+	// ne.DeviceStateProvisioning,
+	// ne.DeviceStateWaitingClusterNodes,
+	// ne.DeviceStateWaitingPrimary,
+	// ne.DeviceStateWaitingSecondary,
+}
+
+var ecxNetworkDeviceSchemaNames = map[string]string{
```

Comment:
`SchemaNames` and `SchemaDescriptions` vars are copied whole-sale from the Network Device Resource.  Could have referenced them directly but not sure if that's a 'bad idea', or not idiomatic etc.

Reply:
Since the virtual device is a product of Network Edge (NE), could you rename the schemaName and schemaDescription to for example neDeviceSchemaNames and neDeviceDescriptions to avoid the confusion?